### ISSUE TITLE
CAF Script Error - check_for_MAXIS vs check_for_password

### ISSUE THREAD

**Issue by Y-Lor-Y** [MN-Script-Team/DHS-MAXIS-Scripts#1409]
There has been two reports that the CAF script is giving them an error at the end on SIR script discussion board.

" CAF SCRIPT ERROR
We are getting an error that says this when we get the the end of the CAF script and try to submit: Script Error at Line 1, Column 1: Type mismatch: 'check_for_password' This happened yesterday and is still happening today."

**Comment by CDPotter** [comment]
I haven't had a chance to run it too much but I haven't run into any errors when running on a test case this morning. I also haven't heard any complaints from any workers here. The comments on SIR are coming from Aikin and Brown counties if that helps any.

**Comment by theVKC** [comment]
This is weird. The check_for_password function is only in beta. Aitkin isn't a beta county: perhaps they are inadvertently running beta scripts.

@IlseFerris, your users are mostly not on the beta. Can you check with them to see if they are seeing this error?
@Y-Lor-Y or @C-Love: was this happening anywhere in Ramsey, beta users or otherwise?

**Comment by theVKC** [comment]
If someone can confirm this is happening in all cases on the CAF script for non-beta agencies, I have an idea for a quick fix. @MN-Script-Team/county-agency-scriptwriters

**Comment by IlseFerris** [comment]
No issue here. I'll see if I can get it to password out, and test that as well.

**Comment by theVKC** [comment]
Can someone ask the reporting counties to check their beta status in Global Variables? If they're set to beta, they should undo that.

**Comment by theVKC** [comment]
Never mind, I'll ask.

**Comment by Y-Lor-Y** [comment]
I've tested both the live region and beta region, no errors found on my end.

**Comment by theVKC** [comment]
I'm leaning toward a beta/release mixup. Another reason it's a good thing to go to scriptwriter-counties-only-on-the-beta if it ends up being the cause.

**Comment by RobertFewins-Kalb** [comment]
I'm confused by the type mismatch error. Why would we be getting a type mismatch error?

**Comment by theVKC** [comment]
Because reasons. Functions are weird sometimes. I've seen this when a function doesn't exist. If you accessed the beta CAF script but the RELEASE funclib, for example.

**Comment by CDPotter** [comment]
Side note: Polled 5 workers here and no problems with the CAF.

**Comment by theVKC** [comment]
Because your folks are BETA, I don't suspect they'd run into it. Try running it as an Aitkin county worker, but manually setting your agency as beta. Perhaps that would yield a result?

**Comment by theVKC** [comment]
(They're x101)

**Comment by CDPotter** [comment]
So I changed the worker_county_code to x101 and left BETA as True on my local globvar and got this message:
![2015-11-18 10_09_59-bluezone script host _ msgbox](https://cloud.githubusercontent.com/assets/7841989/11246260/8cfaf84c-8ddc-11e5-829c-c301d0819273.png)

**Comment by RobertFewins-Kalb** [comment]
...working with @CDPotter on a solution...

**Comment by RobertFewins-Kalb** [comment]
Found it

**Comment by RobertFewins-Kalb** [comment]
Wait...no...poo

**Comment by courtrightd** [comment]
I can recreate the type_mismatch by forcing a local copy of BETA to look at the release functions library, but I can’t spot where one could get to the Beta/Master repository but still have Funclib on release, unless it is something with the IsEmpty(Funclib_URL) = FALSE in the funclib header block.  (somehow they are switched to BETA or master, but still have an old release Funclib loaded?)

**Comment by RobertFewins-Kalb** [comment]
I wonder if it's because the value for beta_agency is stored as a string instead of a Boolean expression... "true" <> true

**Comment by CDPotter** [comment]
When I changed my beta_agency to "true" it let me open notes 0-G (but I didn't get the error when running the CAF)

**Comment by RobertFewins-Kalb** [comment]
Could they be running locally as an agency?

**Comment by theVKC** [comment]
Where in the code is it a string?

**Comment by RobertFewins-Kalb** [comment]
It's not. But if someone is monkeying with the code and puts it in as a string...

**Comment by courtrightd** [comment]
Could we get a copy of their global variables file to look at?

**Comment by RobertFewins-Kalb** [comment]
I'm just going to call the user that reported the error

**Comment by theVKC** [comment]
I doubt they're running locally.

Feel free to ask them via SIR or email. I'm working on a time sensitive report, and have been asked not to investigate this.

**Comment by Y-Lor-Y** [comment]
I downloaded scripts file into a local folder as x101 county...Ran it with Beta = False, I did not encounter any error as far as running the CAF script.  When I change Beta = True, I encountered what Charles stated above of the MsgBox error statement when pressing the "NOTES" button on the Power Pad.

**Comment by RobertFewins-Kalb** [comment]
on the phone with Kaycie...

**Comment by theVKC** [comment]
And that MsgBox @CDPotter reported is intentional error-reduction for me. It's shutting out users without beta. But, it's easy for a scriptwriter to override it.

I am suspecting `IsEmpty(Funclib_URL) = FALSE` might have something to do with it. Perhaps it's missing from their GlobVar.

**Comment by RobertFewins-Kalb** [comment]
Has check_for_password been added to anything else?

**Comment by RobertFewins-Kalb** [comment]
I think I found the problem for Aitkin County...

![image](https://cloud.githubusercontent.com/assets/7842016/11248027/8f3cbc00-8de4-11e5-8ca9-ee9cbf56956d.png)


I am working with the agency to resolve and test.

**Comment by RobertFewins-Kalb** [comment]
So...

Aitkin County's beta_agency value is FALSE, but the code is backward and points the agency to the BETA repository. When the FuncLib is called, it is pointing to the RELEASE FuncLib. The BETA version of the CAF calls the function in question, but that function doesn't exist in RELEASE FuncLib.

We updated the agency's GlobVar. Now it reads IF beta_agency = TRUE THEN ...
It appears that solved the problem. I will reach out to Brown County (Margaret).

Kaycie says the GlobVar was previously updated on 6/3/2015.

**Comment by theVKC** [comment]
Is anyone else's in reverse? That could inform the source of our problem.

And thank you @RobertFewins-Kalb! Above and beyond the call of duty here.

**Comment by theVKC** [comment]
I just checked, and the default is set to TRUE. Which works. So, I have a theory.

Months back, I fixed that BETA issue so that non-BETA agencies would be denied access to scripts if they were incorrectly accessing the BETA: this was done to prevent this very issue.

I issued instructions saying to update `beta_agency = True` to `beta_agency = False`. Perhaps instead of just updating it in the one place, they did a find/replace and replaced all instances, causing this reversal. This explains why the issue is not widespread.
